lib/api/controllers/authController.js

```diff
@@ -430,11 +430,11 @@ class AuthController extends NativeController {
       token = request.input.jwt;
     }
     else {
-      token = request.getBodyString('token');
+      token = request.getBodyString('token', '') || null;
```

Comment:
```suggestion
      token = request.getBodyString('token', '', null);
```

Reply:
@Aschen The second parameter of `getBodyString` is the default value, the goal here is to set the `token` to null if there is no body or the field `token` does not exists.

The catch is that `getBodyString` will throw if the default value is set to null and the field is missing, that's why it's defaulted to an empty string that will then be evaluated with `|| null` to retrieve a null value if the field is missing.

Reply:
Ok I see the problem. Then the modification suggested here should fix that https://github.com/kuzzleio/kuzzle/pull/2237#issuecomment-998654473

Reply:
I read the suggested modifications  and this will not fix the probleme, because if we give a null value to lodash, if not found lodash will return the null value causing the method getBodyString to throw because the value could not be undefined or null.

Reply:
Because we shouldn't throw exception for null value. The `missing_argument` error should only be thrown when the argument is missing

Reply:
Yes, so we should change the method to throw `missing_argument` not based on the returned value of `_.get` but `_.has`
to verify that the argument is not missing